src/protections/consequence.ts

```diff
@@ -0,0 +1,17 @@
+
+
+export enum ConsequenceType {
+    alert,
+    redact,
+    ban
+}
+
+export class Consequence {
+    /*
+     * A description of an action to take when a protection detects abuse
+     *
+     * @param type Action to take
+     * @param reason User-facing explanation of why we're taking an action
```

Comment:
Nit: Which user? Moderators or bystanders? Here, I believe it's moderators.

Also, can it contain html?